server/historian/packages/historian-base/src/routes/git/blobs.ts

```diff
@@ -38,7 +34,7 @@ export function create(
 	const router: Router = Router();
 
 	const tenantThrottleOptions: Partial<IThrottleMiddlewareOptions> = {
-		throttleIdPrefix: (req) => getParam(req.params, "tenantId"),
+		throttleIdPrefix: (req) => req.params.tenantId,
```

Comment:
Any idea why we used getParam for these cases in the first place? Is it possible that the middleware can change its structure to make it an array?

Reply:
I don't know why `getParam` exists honestly. 

```ts
export function getParam(params: Params, key: string) {
	return Array.isArray(params) ? undefined : params[key];
}
```

Even if `params` is an Array, `params[key]` would return `undefined`, so both sides of the ternary are equivalent, making it an unnecessary function.

Regardless, the real issue here is that `throttleIdPrefix` type definition is _slightly_ incorrect.
```ts
// current
throttleIdPrefix?: string | ((req: Request) => string);
// should be
throttleIdPrefix?: string | ((req: Request) => string | undefined);
```

Thankfully, the `getThrottleId` function handles the `falsy` case regardless.

Reply:
Also, in all these cases, `:tenantId` is a required Route path param, so it can't be `undefined`. It's difficult to express that in Types, though.